### DISCUSSION TITLE
[Idea] Guest Orders & New Customer Accounts - Automatically Merge?

### DISCUSSION THREAD

**Discussion by sampsonzak** [PrestaShop/PrestaShop#34904] [Ideas]
### The problem

When I receive guest orders, and the customers then register an account, to view their order details/tracking details at a later date, to find that the order's aren't linked and they can't find any information on their order.

They then email us and create workload for support staff to delete the account they've just made, and to transform their guest order into an account.

### The solution

If a guest order is placed, and an account is registered under the same email address, after the guest order; then add the guest order to the customer account!

This will save lots of confusion, duplicate accounts under the same email address, and overall a much better experience for the customer, and the website owner.

### Do you plan to work on this subject?

- [ ] I'm willing to contribute a [formal specification](https://github.com/PrestaShop/prestashop-specs).
- [ ] I'm willing to provide any wireframes or design assets required for this feature.
- [ ] I'm willing to submit a Pull Request that implements this feature.
- [ ] I'm willing to help verify that the implemented feature works as intended and produces no unintended side effects.

**Comment by HelixMartin** [comment]
Well the problem is that email-adresses can be reused if dropped by somebody else. Creating account let's say two years later can give the new owner of the email access to order information of somebody else. This is a no-no ....

**Comment by darkf3d3** [comment]
The reuse of dropped e-mails should be prohibited. 
Anyone with a reused email address will be able to access to an account using the forgot password feature.
